NPE - From detectProjectState -> compareVersions

I'm upgrading from an older version of OneSignal where all the config was done manually by editing the manifest, added GCM manually, etc. Now I'm on Android Studio 3.0.1 with Gradle 4.4 and com.android.tools.build:gradle:3.1.3 (also I had to enable multidex) and cannot get the plugin installed:
```java
Caused by: java.lang.NullPointerException
        at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.strategy.VersionParser.transform(VersionParser.java:39)
        at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.strategy.VersionParser.transform(VersionParser.java:25)
        at org.gradle.api.Transformer$transform.call(Unknown Source)
        at com.onesignal.androidsdk.GradleProjectPlugin.compareVersions(GradleProjectPlugin.groovy:456)
        at com.onesignal.androidsdk.GradleProjectPlugin$compareVersions$0.callStatic(Unknown Source)
        at com.onesignal.androidsdk.GradleProjectPlugin.detectProjectState(GradleProjectPlugin.groovy:201)
        at com.onesignal.androidsdk.GradleProjectPlugin.apply(GradleProjectPlugin.groovy:167)
        at com.onesignal.androidsdk.GradleProjectPlugin.apply(GradleProjectPlugin.groovy)
        at org.gradle.api.internal.plugins.ImperativeOnlyPluginTarget.applyImperative(ImperativeOnlyPluginTarget.java:42)
        at org.gradle.api.internal.plugins.RuleBasedPluginTarget.applyImperative(RuleBasedPluginTarget.java:50)
        at org.gradle.api.internal.plugins.DefaultPluginManager.addPlugin(DefaultPluginManager.java:165)
        at org.gradle.api.internal.plugins.DefaultPluginManager.access$200(DefaultPluginManager.java:47)
        at org.gradle.api.internal.plugins.DefaultPluginManager$AddPluginBuildOperation.run(DefaultPluginManager.java:252)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:144)
        ... 103 more
```
2 things:
1) other plugins load fine, not sure why this one doesn't
2) I liked doing the config by hand -- I have no idea what your plugin does and would prefer to do it manually, it's not that hard and I would feel better knowing what you are adding to my app.

also:
```gradle
    compileSdkVersion 27
    buildToolsVersion '28.0.1'
    multiDexEnabled true
```

--- a/src/main/groovy/com/onesignal/androidsdk/GradleProjectPlugin.groovy
+++ b/src/main/groovy/com/onesignal/androidsdk/GradleProjectPlugin.groovy
@@ -198,7 +198,7 @@ class GradleProjectPlugin implements Plugin<Project> {
         if (plugin == null)
             return
 
-        if (compareVersions(getAGPVersion(plugin), '3.0.0') == -1) {
+        if (isAGPVersionOlderThan(plugin, '3.0.0')) {
             project.logger.warn("WARNING: The onesignal-gradle-plugin MUST be before com.android.application!")
             project.logger.warn("   Please put onesignal-gradle-plugin first OR update to com.android.tools.build:gradle:3.0.0 or newer!")
 
@@ -208,15 +208,24 @@ class GradleProjectPlugin implements Plugin<Project> {
         }
     }
 
+    static boolean isAGPVersionOlderThan(Plugin plugin, String version) {
+        def agpVersion = getAGPVersion(plugin)
+        if (!agpVersion) {
+            if (project)
+                project.logger.warn('OneSignal Warning: Could not get AGP plugin version')
+            return false
+        }
+
+        compareVersions(agpVersion, version) == -1
+    }
+
     static String getAGPVersion(Plugin plugin) {
         try {
             def cl = plugin.class.classLoader as URLClassLoader
             def inputStream = cl.findResource('META-INF/MANIFEST.MF').openStream()
             def manifest = new Manifest(inputStream)
             return manifest.mainAttributes.getValue('Plugin-Version')
-        } catch (ignore) {
-            project.logger.warn("OneSignal Warning: Could not get AGP plugin version")
-        }
+        } catch (ignore) {}
         null
     }
 

--- a/src/test/groovy/com/onesignal/androidsdk/MainTest.groovy
+++ b/src/test/groovy/com/onesignal/androidsdk/MainTest.groovy
@@ -246,6 +246,11 @@ class MainTest extends Specification {
         assert GradleProjectPlugin.lowerMaxVersion("[26.0.0, 27.1.0)", "27.+") == '[26.0.0, 27.1.0)'
     }
 
+    def 'isAGPVersionOlderThan should thrown when getAGPVersion is null'() {
+        given:
+        GradleProjectPlugin.isAGPVersionOlderThan(null, '3.0.0')
+    }
+
     def "Upgrade to compatible OneSignal SDK when targetSdkVersion is 26"() {
         when:
         def results = runGradleProject([

